python_modules/dagster/dagster_tests/core_tests/test_logging.py

```diff
@@ -154,6 +152,85 @@ def test_multiline_logging_complex():
     assert captured_results[0].split("\n") == expected_results
 
 
+def _setup_test_two_handler_log_mgr():
+    test_formatter = logging.Formatter(fmt='%(levelname)s :: %(message)s')
+
+    test_info_handler = logging.StreamHandler(sys.stdout)
+    test_info_handler.setLevel("INFO")
+    test_info_handler.setFormatter(test_formatter)
+
+    test_warn_handler = logging.StreamHandler(sys.stdout)
+    test_warn_handler.setLevel("WARN")
+    test_warn_handler.setFormatter(test_formatter)
+
+    dl = DagsterLogManager(
+        DagsterLoggingMetadata(run_id="123"),
+        [],
+        [test_info_handler, test_warn_handler],
+    )
+    return dl
+
+
+def test_handler_in_log_manager(capsys):
+    dl = _setup_test_two_handler_log_mgr()
+
+    dl.info("test")
+    dl.warning("test")
+
+    out, _ = capsys.readouterr()
+
+    assert re.search(r"INFO :: system - 123 - test", out)
+    assert len(re.findall(r"WARNING :: system - 123 - test", out)) == 2
+
+
+def test_handler_in_log_manager_with_tags(capsys):
+    dl = _setup_test_two_handler_log_mgr()
+    dl = dl.with_tags(**{"pipeline_name": "test_pipeline"})
+
+    dl.info("test")
+    dl.warning("test")
+
+    out, _ = capsys.readouterr()
+
+    assert re.search(r"INFO :: test_pipeline - 123 - test", out)
+    assert len(re.findall(r"WARNING :: test_pipeline - 123 - test", out)) == 2
+
+
+def test_capture_handler_log_records():
+    class TestHandler(logging.Handler):
+        def __init__(self):
+            self.captured = []
+            super().__init__(logging.INFO)
+
+        def emit(self, record):
+            self.captured.append(record)
+
+    capture_handler = TestHandler()
+
+    dl = DagsterLogManager(
+        logging_metadata=DagsterLoggingMetadata(
+            run_id="123456", pipeline_name="pipeline", step_key="some_step"
+        ),
+        loggers=[],
+        handlers=[capture_handler],
+    )
+
+    dl.info("info")
+    dl.critical("critical error")
```

Comment:
try dl.critical("critical error", extra={"foo":"bar"}), with a corresponding `assert captured_critical_record.foo == "bar"`. I believe this assertion will fail.